pkg/scheduler/framework/runtime/framework.go

```diff
@@ -463,23 +490,52 @@ func (f *frameworkImpl) expandMultiPointPlugins(profile *config.KubeSchedulerPro
 			// the user intent is to override the default plugin or make some other explicit setting.
 			// Either way, discard the MultiPoint value for this plugin.
 			// This maintains expected behavior for overriding default plugins (see https://github.com/kubernetes/kubernetes/pull/99582)
-			if enabledSet.Has(ep.Name) {
+			if enabledSet.has(ep.Name) {
+				overridePlugins.insert(ep.Name)
 				klog.InfoS("MultiPoint plugin is explicitly re-configured; overriding", "plugin", ep.Name)
 				continue
 			}
 
 			// if this plugin is already registered via MultiPoint, then this is
 			// a double registration and an error in the config.
-			if multiPointEnabled.Has(ep.Name) {
+			if multiPointEnabled.has(ep.Name) {
 				return fmt.Errorf("plugin %q already registered as %q", ep.Name, pluginType.Name())
 			}
 
 			// we only need to update the multipoint set, since we already have the specific extension set from above
-			multiPointEnabled.Insert(ep.Name)
+			multiPointEnabled.insert(ep.Name)
 
 			newPlugins := reflect.Append(plugins, reflect.ValueOf(pg))
 			plugins.Set(newPlugins)
 		}
+
+		// Reorder plugins if needed. Here is the expected order:
```

Comment:
It seems like we rearrange all the plugin orders here. So when ` len(profile.Plugins.MultiPoint.Enabled) > 0`, we no longer need above `updatePluginList` process anymore, right?  If this makes sense, we should merge them together.  https://github.com/kubernetes/kubernetes/blob/e9af39914797d40766b166e18dac3a441a2bf5d4/pkg/scheduler/framework/runtime/framework.go#L334-L338

Reply:
I still prefer to keep them separate. When `len(profile.Plugins.MultiPoint.Enabled) > 0`, it's not necessarily going to rearrange the plugins - e.g., all plugins defined in multipoint has been specified in the regular extension point.